src/Features/Core/Portable/FindUsages/DefinitionItem.DetachedDefinitionItem.cs

```diff
@@ -76,18 +77,22 @@ public override int GetHashCode()
         public async Task<DefaultDefinitionItem?> TryRehydrateAsync(Solution solution, CancellationToken cancellationToken)
         {
             using var converted = TemporaryArray<DocumentSpan>.Empty;
+            using var convertedClassifiedSpans = TemporaryArray<ClassifiedSpansAndHighlightSpan?>.Empty;
             foreach (var ss in SourceSpans)
             {
                 var documentSpan = await ss.TryRehydrateAsync(solution, cancellationToken).ConfigureAwait(false);
                 if (documentSpan == null)
                     return null;
 
                 converted.Add(documentSpan.Value);
+
+                // todo: consider serializing this data.
```

Comment:
serialization never serialized out the properties in a DocumentSpan (how could it, it just saw a dictionary with objects).  So this is no change from before. 

Now that the data is strongly typed and simple structs, we could consdier marshalling.